src/test/java/net/sf/jabref/logic/autocompleter/BibtexKeyAutoCompleterTest.java

```diff
@@ -0,0 +1,153 @@
+package net.sf.jabref.logic.autocompleter;
+
+import static org.mockito.Mockito.*;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import net.sf.jabref.model.entry.BibtexEntry;
+
+public class BibtexKeyAutoCompleterTest {
+
+    @Test(expected = NullPointerException.class)
+    public void initAutoCompleterWithNullPreferenceThrowsException() {
+        new BibtexKeyAutoCompleter(null);
+    }
+
+    @Test
+    public void completeWithoutAddingAnythingReturnsNothing() {
+        AutoCompletePreferences preferences = mock(AutoCompletePreferences.class);
+        BibtexKeyAutoCompleter autoCompleter = new BibtexKeyAutoCompleter(preferences);
+
+        String[] result = autoCompleter.complete("test");
+        Assert.assertEquals(0, result.length);
```

Comment:
Hm, normally, I do not use arrays but lists instead. Then, I get a detailed message comparing two lists by Assert.assertEquals(Collections.emptyList(), actualList);